src/Microsoft.Health.Fhir.Shared.Api/Features/Filters/OperationOutcomeExceptionFilterAttribute.cs

```diff
@@ -158,6 +158,34 @@ public override void OnActionExecuted(ActionExecutedContext context)
                         break;
                 }
 
+                context.Result = healthExceptionResult;
+                context.ExceptionHandled = true;
+            }
+            else if (context.Exception is SystemException systemException)
+            {
+                OperationOutcomeResult healthExceptionResult;
+
+                switch (systemException)
+                {
+                    case NotImplementedException ex:
```

Comment:
We could probably change the `NotImplmplementedException` that we're currently throwing in the handler to a `MethodNotAllowedException`. That way this particular change wouldn't be required. `MethodNotAllowedException` is the type of exception we're using for when things like conditional create/update.